File: pinyougou_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

/**
 * 品牌
 */
public interface BrandService {
//    查询
    List<TbBrand> findAll();
//    分页
    PageResult findPage(int page,int rows);
//    添加
    void add(TbBrand brand);
//    根据id查询
    TbBrand findOne(Long id);
//    修改
    void update(TbBrand brand);
//    删除
    void delete(Long[] ids);
//    条件查询及分页
    PageResult findPage(TbBrand brand,int page,int rows);
//    品牌下拉框数据
    List<Map> selectOptionList();
}

File: pinyougou_manager_web/src/main/webapp/js/service/brandService.js
//品牌服务层
//把$http内置服务的请求路劲抽取出来
app.service('brandService',function ($http) {
        //this就是上面的参数$http
        //查询品牌列表
        this.findAll=function () {
            return $http.get('../brand/findAll.do');
        }

        //分页
        this.page=function (page,rows) {
            return $http.get('../brand/findPage.do?page='+page +'&rows='+rows);
        }

        //增加
        this.add=function (entity) {
            return $http.post('../brand/add.do',entity);
        }

        //修改
        this.update=function (entity) {
            return $http.post('../brand/update.do',entity);
        }

    //根据id查询
        this.findOne=function (id) {
            return  $http.get('../brand/findOne.do?id='+id);
        }

    //删除
        this.dele=function (ids) {
            return  $http.get('../brand/delete.do?ids='+ids);
        }

    //条件查询
        this.search=function (page,rows,searchEntity) {
            return $http.post('../brand/search.do?page='+page +'&rows='+rows, searchEntity);
        }
    //下拉列表数据
        this.selectOptionList=function () {
            return $http.get('../brand/selectOptionList.do');
        }
})